tensorzero-internal/src/clickhouse/migration_manager/migrations/migration_0020.rs

```diff
@@ -0,0 +1,278 @@
+                id: "0020".to_string(),
+                message: "One or more of the inference datapoint tables do not exist".to_string(),
+            }));
+        }
+
+        Ok(())
+    }
+
+    async fn should_apply(&self) -> Result<bool, Error> {
+        let tag_inference_table_exists =
+            check_table_exists(self.clickhouse, "TagInference", "0020").await?;
+        let tag_chat_inference_view_exists =
+            check_table_exists(self.clickhouse, "TagChatInferenceView", "0020").await?;
+        let tag_json_inference_view_exists =
+            check_table_exists(self.clickhouse, "TagJsonInferenceView", "0020").await?;
+        let chat_inference_datapoint_staled_at_column_exists = check_column_exists(
+            self.clickhouse,
+            "ChatInferenceDatapoint",
+            "staled_at",
+            "0020",
+        )
+        .await?;
+        let json_inference_datapoint_staled_at_column_exists = check_column_exists(
+            self.clickhouse,
+            "JsonInferenceDatapoint",
+            "staled_at",
+            "0020",
+        )
+        .await?;
+        let chat_default_updated_at = get_default_expression(
+            self.clickhouse,
+            "ChatInferenceDatapoint",
+            "updated_at",
+            "0020",
+        )
+        .await?;
+        let chat_default_updated_at_correct = chat_default_updated_at == "now64()";
+        let json_default_updated_at = get_default_expression(
+            self.clickhouse,
+            "JsonInferenceDatapoint",
+            "updated_at",
+            "0020",
+        )
+        .await?;
+        let json_default_updated_at_correct = json_default_updated_at == "now64()";
+        Ok(!tag_inference_table_exists
+            || !tag_chat_inference_view_exists
+            || !tag_json_inference_view_exists
+            || !chat_inference_datapoint_staled_at_column_exists
+            || !json_inference_datapoint_staled_at_column_exists
+            || !chat_default_updated_at_correct
+            || !json_default_updated_at_correct)
+    }
+
+    async fn apply(&self) -> Result<(), Error> {
+        // Only gets used when we are not doing a clean start
+        let view_offset = Duration::from_secs(15);
+        let view_timestamp = (std::time::SystemTime::now()
+            .duration_since(std::time::UNIX_EPOCH)
+            .map_err(|e| {
+                Error::new(ErrorDetails::ClickHouseMigration {
+                    id: "0020".to_string(),
+                    message: e.to_string(),
+                })
+            })?
+            + view_offset)
+            .as_secs();
+
+        let query = r#"
+            CREATE TABLE IF NOT EXISTS TagInference
+                (
+                    key String,
+                    value String,
+                    function_name LowCardinality(String),
+                    variant_name LowCardinality(String),
+                    episode_id UUID,
+                    inference_id UUID,
```

Comment:
do we want this to be stored as a UInt128? Might be useful for getting things in order by inference id